File: Sources/Sentry/SentrySDKSettings.m
#import "SentrySDKSettings.h"
#import "SentryOptions.h"
#import "SentrySDKInternal.h"

@implementation SentrySDKSettings

- (instancetype)initWithOptions:(SentryOptions *_Nullable)options
{
    if (self = [super init]) {
        _autoInferIP = options.sendDefaultPii;
    }

    return self;
}

- (instancetype)initWithDict:(NSDictionary *)dict
{
    if (self = [super init]) {
        if ([dict[@"infer_ip"] isKindOfClass:[NSString class]]) {
            _autoInferIP = [dict[@"infer_ip"] isEqualToString:@"auto"];
        }
    }
    return self;
}

- (nonnull NSDictionary<NSString *, id> *)serialize
{
    return @{ @"infer_ip" : _autoInferIP ? @"auto" : @"never" };
}

@end

File: Sources/Swift/Protocol/SentrySDKSettings.swift
/**
 * Describes the settings for the Sentry SDK
 * @see https://develop.sentry.dev/sdk/event-payloads/sdk/
 */
struct SentrySDKSettings {
    
    init() {
        autoInferIP = false
    }

    init(autoInferIP: Bool) {
        self.autoInferIP = autoInferIP
    }
    
    init(dict: NSDictionary) {
        if let inferIp = dict["infer_ip"] as? String {
            autoInferIP = inferIp == "auto"
        } else {
            autoInferIP = false
        }
    }
    
    let autoInferIP: Bool
    
    func serialize() -> NSDictionary {
        [
            "infer_ip": autoInferIP ? "auto" : "never"
        ]
    }
}
